components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -103,7 +103,9 @@ public String jwks() {
             String errorMessage = "Error while generating the keyset for tenant domain: " + tenantDomain;
             return logAndReturnError(errorMessage, e);
         } finally {
-            FrameworkUtils.endTenantFlow();
+            if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
```

Comment:
This fix this perfectly fine.

I was wondering whether we could get rid of the if block by,

#1 - Move the validating tenant ID logic outside the try block.
#2 - Start the tenant flow at the beginning of the try block (regardless of whatever the tenant it is)
#3 - end tenant flow at the finally block

That would make this code block readable and less error-prone. WDYT?

Reply:
Also even, after the proposed fix we can face the same issue if 
`IdentityTenantUtil.getTenantId(tenantDomain)` line throws an exception which would result in finally block to attempt to end a tenant flow that was never started.

Reply:
@mefarazath That occurred to me. It looks better; however, I was wondering about the possibility of it breaking something since the original code had put it specifically inside the block for non-super tenants.

Reply:
If it works for super tenant and tenant endpoints we are good.